src/driver.ts

```diff
@@ -262,20 +229,14 @@ export abstract class CMakeDriver implements vscode.Disposable {
     if (this._kit.environmentVariables) {
       util.objectPairs(this._kit.environmentVariables).forEach(([k, v]) => this._kitEnvironmentVariables.set(k, v));
     }
-    switch (this._kit.type) {
```

Comment:
I like this PR, it reduces the complexity of the code and increases the flexibility for the user.